packages/spark/sodasql/dialects/spark_dialect.py

```diff
@@ -57,6 +57,9 @@ def sql_tables_metadata_query(self, limit: Optional[int] = None, filter: str = N
 
     def sql_tables_metadata(self, limit: str = 10, filter: str = None):
         # TODO Implement limit
+        if self.database is None:
+            raise NotImplementedError("Can not query for tables when database is not given.")
+
         with self.create_connection().cursor() as cursor:
```

Comment:
Might need to add a Try / Exception here as the connection might fail for multiple reasons

Reply:
Sure, I think we solve this with #479, then we always use the `Warehouse` class to query the ware house. In that class we can implement this properly with a `try-except`.